### PR TITLE
Add section for validating models in the Model page

### PR DESCRIPTION
**Pull request by TomeHirata**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/TomeHirata/mlflow/pull/13988?quickstart=1)

#### Install mlflow from this PR

```
# Use `%sh` to run this command on Databricks
OPTIONS=$(if pip freeze | grep -q 'mlflow @ git+https://github.com/mlflow/mlflow.git'; then echo '--force-reinstall --no-deps'; fi)
pip install $OPTIONS git+https://github.com/mlflow/mlflow.git@refs/pull/13988/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 13988
```

</p>
</details>

### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?

Validating models before deployment is an important step, and we want to enhance the documentation for the process. This PR introduces a new section in the Model page describing the way to use mlflow.models.predict for this purpose.

Screenshot 2024-12-05 at 18 10 29

### How is this PR tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [ ] No. You can skip the rest of this section.
- [x] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [x] Instructions

#### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Should this PR be included in the next patch release?

`Yes` should be selected for bug fixes, documentation updates, and other small changes. `No` should be selected for new features and larger changes. If you're unsure about the release classification of this PR, leave this unchecked to let the maintainers decide.

<details>
<summary>What is a minor/patch release?</summary>

- Minor release: a release that increments the second part of the version number (e.g., 1.2.0 -> 1.3.0).
  Bug fixes, doc updates and new features usually go into minor releases.
- Patch release: a release that increments the third part of the version number (e.g., 1.2.0 -> 1.2.1).
  Bug fixes and doc updates usually go into patch releases.

</details>

- [ ] Yes (this PR will be cherry-picked and included in the next patch release)
- [x] No (this PR will be included in the next minor release)

### CODE REVIEW

File: docs/source/models.rst [hunk 1]
```diff
@@ -3918,6 +3918,25 @@ back to ``numpy ndarray`` type as required by ``sktime`` inference API.
     response = requests.post(url, json=json_data)
     print(f"\nPyfunc 'predict_interval':\n${response.json()}")
 
+
+Validate Models before Deployment
+---------------------------------
+
+After logging your model with MLflow Tracking, it is recommended to validate the model locally before deploying it to production.
+The :py:func:`mlflow.models.predict()` API offers a functionality for testing your model in a virtual environment.
+You can use this API by passing a sample input, as shown in the example code below.
```
Comment (serena-ruan) [inline docs/source/models.rst:3927]:
```suggestion
The :py:func:mlflow.models.predict() API enables you to test your model in a virtual environment for isolated execution. You can utilize this functionality by passing a sample input, as demonstrated in the example code below.
```

File: docs/source/models.rst [hunk 2]
```diff
@@ -3918,6 +3918,25 @@ back to ``numpy ndarray`` type as required by ``sktime`` inference API.
...
+Additionally, this API is helpful for validating the environment configuration. 
+For more details, refer to  :ref:`Validating Environment for Prediction <validating-environment-for-prediction>`.
+
+.. code-block:: python
+
+    import mlflow
+
+    mlflow.models.predict(
+        model_uri=f"runs:/{run_id}/model",
```
Comment (serena-ruan) [inline docs/source/models.rst:3936]:
```suggestion
        model_uri=model_uri,
```
As the model path could be different than 'model'. I'll update the API docstring separately.

**Review by serena-ruan — APPROVED** [review]
LGTM with minor comments!

File: docs/source/models.rst [hunk 3]
```diff
@@ -3918,6 +3918,25 @@ back to ``numpy ndarray`` type as required by ``sktime`` inference API.
...
     print(f"\nPyfunc 'predict_interval':\n${response.json()}")
 
+
+Validate Models before Deployment
+---------------------------------
+
+After logging your model with MLflow Tracking, it is recommended to validate the model locally before deploying it to production.
+The :py:func:mlflow.models.predict() API enables you to test your model in a virtual environment for isolated execution. 
+You can utilize this functionality by passing a sample input, as demonstrated in the example code below.
+Additionally, this API is helpful for validating the environment configuration.
```
Comment (harupy) [inline docs/source/models.rst:3928]: `the environment configuration` sounds a bit ambiguous. Can we elaborate on it more?

File: docs/source/models.rst [hunk 4]
```diff
@@ -3918,6 +3918,25 @@ back to ``numpy ndarray`` type as required by ``sktime`` inference API.
...
+After logging your model with MLflow Tracking, it is recommended to validate the model locally before deploying it to production.
+The :py:func:mlflow.models.predict() API enables you to test your model in a virtual environment for isolated execution. 
+You can utilize this functionality by passing a sample input, as demonstrated in the example code below.
+Additionally, this API is helpful for validating the environment configuration. 
+For more details, refer to  :ref:`Validating Environment for Prediction <validating-environment-for-prediction>`.
+
+.. code-block:: python
+
+    import mlflow
+
```
Comment (harupy) [inline docs/source/models.rst:3934]: Is it possible to add a minimum code to log a model?

File: docs/source/models.rst (hunk 3 above)
Comment (TomeHirata) [reply docs/source/models.rst:3928]: Sure, that sounds better, let me change it here.

File: docs/source/models.rst (hunk 4 above)
Comment (TomeHirata) [reply docs/source/models.rst:3934]: Added a bit more codes, does this look good to you?

Comment (TomeHirata) [reply docs/source/models.rst:3781]: Sorry, forgot to push the change to this repo. Will address it in another PR.

**Pull request merged**